### PR TITLE
Add CI to push container image

### PR DESCRIPTION
**Pull request by onprem**
This PR adds a GH action that will build and push latest container image to `quay.io/observatorium/rules-objstore` on every merge to `main`.

### CODE REVIEW

File: .github/workflows/push-image.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          install: true
+
+      - name: Login to DockerHub
```
Comment (matej-g) [inline .github/workflows/push-image.yml:24]: Nit:
```suggestion
      - name: Login to Quay.io
```

File: .github/workflows/push-image.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    name: Go build
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```
Comment (matej-g) [inline .github/workflows/push-image.yml:20]: To make it clear for myself, are we introducing these actions to enable multi-platform build?

Comment (onprem) [reply .github/workflows/push-image.yml:20]: The `qemu` step enables us to do multi platform build in future if we want to. We are not doing cross builds now though. The buildx step however is just the modern equivalent of `docker build` IIRC.

Comment (matej-g) [reply .github/workflows/push-image.yml:20]: I'm just wondering if we should have it (at least the QEMU action), if we're not using it currently, what do you think? I think we can always add it later when we decide to support multi-platform build. I believe we also do not support multi-platform build on other components e.g. API, but we would like to in the future.

**Review by matej-g — APPROVED** [review]
Thanks, looking great :rocket:

**Pull request merged**